### PR TITLE
Integrate both llvm-project@2083e97e (+1 :leftwards_arrow_with_hook:, +1 :cherries:) and torch-mlir@bce800a3

### PR DESCRIPTION
**Pull request by bjacob**
* torch-mlir integrated at bce800a3.
* llvm-project integrated at 2083e97e plus local changes:
    * Reverted https://github.com/llvm/llvm-project/pull/89131 locally: while this change is good in its own right, the `vector.interleave` that it generates (instead of `vector.shuffle`) are not handled by some GPU codegen lowerings.
        * Filed https://github.com/iree-org/iree/issues/17346.
    * Cherry-picked Bazel build fix: https://github.com/llvm/llvm-project/pull/91654
* Several e2e tests have been temporarily disabled, follow-up work is needed to reenable them: https://github.com/iree-org/iree/issues/17344

### CODE REVIEW

**Comment by AmosLewis** [comment]
Could you bump torch-mlir to one more commit forward. This will unblocked a lot SharkTestsuite ONNX model tests, which has a deadline this week.
commit ec6d7aa5d28f110aa5b893e16e502e6198988801
Author: aldesilv <35313749+aldesilv@users.noreply.github.com>
Date:   Wed May 8 14:35:03 2024 -0700

    OnnxToTorch lowering resize op (#3013)

**Comment by AmosLewis** [comment]
Why not bump iree llvmproject into dabdec1001dc368373dd581cf72f37a440873ce? same as the [torch-mlir bce800a3](https://github.com/llvm/torch-mlir/commit/bce800a3f4200f8a28e844a91c2be2d2ce71c174) bring in.

File: third_party/llvm-project
Comment (ScottTodd) [inline third_party/llvm-project:1]: The `opt-125M` failures on https://github.com/iree-org/iree/actions/runs/9020826623/job/24787542994?pr=17330#step:9:39 are odd:
```
___________ IREE compile and run: opt-125M::gpu_vulkan_real_weights ____________
Error invoking iree-compile
Error code: 1
Stderr diagnostics:
opt-125M.mlirbc:0:0: error: attempting to parse a byte at the end of the bytecode
opt-125M.mlirbc:0:0: note: in bytecode version 6 produced by: MLIR19.0.0git


Invoked with:
  cd /home/esaimana/actions-runner/_work/iree/iree/SHARK-TestSuite/iree_tests/pytorch/models/opt-125M && iree-compile opt-125M.mlirbc --iree-hal-target-backends=vulkan-spirv -o opt-125M_gpu_vulkan_real_weights.vmfb
```

I don't think that file has been modified and the test should be passing before these changes. Did the bytecode format or its parsing change?

**Comment by bjacob** [comment]
@AmosLewis 

> Could you bump torch-mlir to one more commit forward. 

OK because it's just one more commit (we need this process to converge now).

> Why not bump iree llvmproject into dabdec1001dc368373dd581cf72f37a440873ce? same as the [torch-mlir bce800a3](https://github.com/llvm/torch-mlir/commit/bce800a3f4200f8a28e844a91c2be2d2ce71c174) bring in.

Because IREE requires additional fixes.

File: third_party/llvm-project
Comment (bjacob) [reply third_party/llvm-project:1]: This is still happening. @MaheshRavishankar @ScottTodd  what do we do, can we check if a MLIR bytecode version bump did just occur and can we then regenerate or temporarily disable this test?

Comment (ScottTodd) [reply third_party/llvm-project:1]: I don't see any recent changes in https://github.com/llvm/llvm-project/tree/main/mlir/include/mlir/Bytecode or https://github.com/llvm/llvm-project/tree/main/mlir/lib/Bytecode. Not sure why just that one test would be failing, so would like to investigate/debug.

To keep the integrate train rolling, we can disable by marking as an expected compile failure in the `_models_` files at https://github.com/iree-org/iree/tree/main/build_tools/pkgci/external_test_suite

Comment (MaheshRavishankar) [reply third_party/llvm-project:1]: I thought vulkan was always failing for sdxl?

Comment (ScottTodd) [reply third_party/llvm-project:1]: Oh the resnet test is also failing. Those (opt-125M and resnet50) are two real (new) failures in parsing the bytecode. The SDXL Vulkan failures are already marked XFAIL, ignore those.

Comment (MaheshRavishankar) [reply third_party/llvm-project:1]: Lets wait for the CI to run again. This might be related to the `vector.shuffle/vector.interleave` issue

Comment (MaheshRavishankar) [reply third_party/llvm-project:1]: Oh no this is bad! It is failing on rocm now, and maybe the error is misleading, but is it failing to read the input file?

**Comment by MaheshRavishankar** [comment]
There are some other tests also failing with the same bytecode parsing error.

**Comment by ScottTodd** [comment]
> There are some other tests also failing with the same bytecode parsing error.

https://github.com/iree-org/iree/actions/runs/9023942190/job/24797359659?pr=17330#step:7:720
```
/work/build-e2e-test-artifacts/e2e_test_artifacts/model_Falcon7bGptqPT.mlirbc:0:0: error: attempting to parse a byte at the end of the bytecode
/work/build-e2e-test-artifacts/e2e_test_artifacts/model_Falcon7bGptqPT.mlirbc:0:0: note: in bytecode version 6 produced by: MLIR18.0.0git
```

Okay phew (or eek), not just files that I uploaded myself. That rules out some things. Let's try to find whatever upstream change(s) affected the parsing.

**Comment by ScottTodd** [comment]
Might have a CUDA test newly timing out: https://github.com/iree-org/iree/actions/runs/9023942176/job/24797058825?pr=17330 had to cancel that job after 6 hours.

**Comment by bjacob** [comment]
> @AmosLewis
> 
> > Could you bump torch-mlir to one more commit forward.
> 
> OK because it's just one more commit (we need this process to converge now).

Actually this (https://github.com/llvm/torch-mlir/pull/3013) turned out to cause additional problems here: https://github.com/iree-org/iree/issues/17345. This integrate has been reverted to its original torch-mlir target.

Discord conversation:
https://discord.com/channels/689900678990135345/1080178290188374049/1238530135444029574

**Review by ScottTodd — APPROVED** [review]
LGTM when CI goes green (assuming also that benchmarks don't have any glaring regressions)

**Pull request merged**
